drivers/SmartThings/zwave-switch/src/test/test_zooz_zen_30_dimmer_relay.lua

```diff
@@ -0,0 +1,748 @@
+  function()
+    test.timer.__create_and_queue_test_time_advance_timer(1, "oneshot")
+    test.socket.capability:__queue_receive({
+      mock_zooz_zen_30_dimmer_relay.id,
+      { capability = "switch", component = "switch1", command = "off", args = {} }
+    })
+    test.socket.zwave:__expect_send(
+      zw_test_utils.zwave_test_build_send_command(
+        mock_zooz_zen_30_dimmer_relay,
+        SwitchBinary:Set({
+          target_value = SwitchBinary.value.OFF_DISABLE
+        },
+        {
+          encap = zw.ENCAP.AUTO,
+          src_channel = 0,
+          dst_channels = {1}
+        }
+        )
+      )
+    )
+    test.wait_for_events()
+    test.mock_time.advance_time(1)
+    test.socket.zwave:__expect_send(
+      zw_test_utils.zwave_test_build_send_command(
+        mock_zooz_zen_30_dimmer_relay,
+        SwitchBinary:Get({},
+          {
+            encap = zw.ENCAP.AUTO,
+            src_channel = 0,
+            dst_channels = {1}
+          })
+      )
+    )
+  end
+)
+
+test.register_coroutine_test(
+  "SwitchLevel capability setLevel commands should evoke the correct Z-Wave SETs and GETs",
+  function ()
+    test.timer.__create_and_queue_test_time_advance_timer(1, "oneshot")
+    test.socket.capability:__queue_receive({
+      mock_zooz_zen_30_dimmer_relay.id,
+      { capability = "switchLevel", component = "main", command = "setLevel", args = { 10 } }
+    })
+    test.socket.capability:__expect_send(
+      mock_zooz_zen_30_dimmer_relay:generate_test_message(
+        "main", capabilities.switch.switch.on()
+      )
+    )
+    test.socket.zwave:__expect_send(
+      zw_test_utils.zwave_test_build_send_command(
+        mock_zooz_zen_30_dimmer_relay,
+        SwitchMultilevel:Set({
+          value = 10,
+          duration = constants.DEFAULT_DIMMING_DURATION
+        })
+      )
+    )
+    test.wait_for_events()
+    test.mock_time.advance_time(1)
+    test.socket.zwave:__expect_send(
+      zw_test_utils.zwave_test_build_send_command(
+        mock_zooz_zen_30_dimmer_relay,
+        SwitchMultilevel:Get({})
+      )
+    )
+  end
+)
+
+test.register_message_test(
+  "Z-Wave SwitchMultilevel reports with non-zero values should evoke Switch and Switch Level capability events",
+  {
+    {
+      channel = "zwave",
+      direction = "receive",
+      message = {
+        mock_zooz_zen_30_dimmer_relay.id,
+        zw_test_utils.zwave_test_build_receive_command(
+          SwitchMultilevel:Report({
+            current_value = 0,
+            target_value = 50,
+            duration = 0
+          })
+        )
+      }
+    },
+    {
+      channel = "capability",
+      direction = "send",
+      message = mock_zooz_zen_30_dimmer_relay:generate_test_message("main", capabilities.switch.switch.on())
+    },
+    {
+      channel = "capability",
+      direction = "send",
+      message = mock_zooz_zen_30_dimmer_relay:generate_test_message("main", capabilities.switchLevel.level(50))
+    }
+  }
+)
+
```

Comment:
@NikitaKuchta You might add unit tests for BasicReport and SwitchBinaryReport  - for both endpoints. 
Test for endpoint 0 should check if switch and switch level events are sent.
Test for endpoint 1 should check if switch event is sent.